src/dxtbx/model/beam.py

```diff
@@ -310,9 +310,20 @@ def imgCIF_H(cbf_handle):
             0.0,
         )
 
+        # and the flux if available
+        try:
+            cbf_handle.find_category(b"diffrn_radiation")
+            cbf_handle.find_column(b"beam_flux")
+            flux = cbf_handle.get_value()
+        except Exception as e:
+            if str(e).split()[-1] != "CBF_NOTFOUND":
```

Comment:
Suspect that there are idiomatically better ways to catch this?

Reply:
Maybe, but we are limited by the SWIG implementation here.  I looked for a 'check for column' function and didn't find it.  The find category/find column patterns seems to be what's available.  When these functions fail, in C they return an error code, but in python they raise a generic exception with a known string.